src/plugins/intel_gpu/include/intel_gpu/runtime/device.hpp

```diff
@@ -19,6 +20,9 @@ struct device {
     virtual memory_capabilities get_mem_caps() const = 0;
 
     virtual bool is_same(const device::ptr other) = 0;
+    virtual bool try_kernel_execution(kernel::ptr kernel) = 0;
+    virtual int8_t get_subgroup_local_block_io_supported() = 0;
+    virtual void set_subgroup_local_block_io_supported(bool support) = 0;
```

Comment:
Adding 3 virtual methods into the device class to implement WA for driver bug is weird...

To my mind the required logic is quite simple:
- device checks extension support only and reports it via device_info
- each program has internal local_block_io_support flag
- the flag is initialized by value in device info
- on program build we adjust flag value by compiling and running simple kernel (which is cached)
- the kernel is not compiled and not run if the extension is not supported
- overhead on running simple kernel on each program creation is expected to be negligible 

Or alternatively we can consider adding 1 virtual method like `validate_device_info()` which will be called once we're sure that cache directory is known

Reply:
I have one question. How about a case not cached?
Does each program compile a simple kernel to adjust its flag?

Reply:
Applied

Reply:
@vladimir-paramuzov  logics in device was removed. Now, it checks extension only.
The flag moved to program.